integration/shareddata/scalars.go

```diff
@@ -353,4 +348,9 @@ func init() {
 
 	must.BeTrue(-doubleMaxPrec != -doubleMaxPrec-1)
 	must.BeTrue(-doubleMaxPrec-1 == -doubleMaxPrec-2)
+
+	must.BeTrue(doubleMaxOverflow > float64(math.MaxInt64))
+	must.BeTrue(doubleMinOverflow <= float64(math.MinInt64))
```

Comment:
```suggestion
	must.BeTrue(doubleMinOverflow < float64(math.MinInt64))
```

Otherwise, we don't check much